src/main/java/com/saralein/server/Main.java

```diff
@@ -7,18 +7,26 @@
 import com.saralein.server.router.ServerRouter;
 import com.saralein.server.server.Server;
 import com.saralein.server.server.SetupServer;
+import com.saralein.server.validation.ArgsValidation;
+import java.io.File;
 
 public class Main {
     public static void main(String[] args) {
+        ArgsValidation argsValidation = new ArgsValidation(System.getProperty("user.home"));
         Logger logger = new ConnectionLogger(System.out);
-        Runtime runtime = Runtime.getRuntime();
 
-        SysFileHelper sysFileHelper = new SysFileHelper("public");
-        ServerRouter router = new ServerRouter(sysFileHelper);
+        if (argsValidation.argsAreValid(args)) {
+            Integer port = argsValidation.parsePort();
+            File root = argsValidation.determineRoot();
+            Runtime runtime = Runtime.getRuntime();
+            SysFileHelper sysFileHelper = new SysFileHelper(root);
+            ServerRouter router = new ServerRouter(sysFileHelper);
+            RequestParser requestParser = new RequestParser();
+            Server server = new SetupServer(logger, runtime, router, requestParser).setup(port);
 
-        RequestParser requestParser = new RequestParser();
-
-        Server server = new SetupServer(logger, runtime, router, requestParser).setup(args);
-        server.run();
+            server.run();
+        } else {
+            logger.log("Invalid arguments: Server cannot run with given arguments.");
```

Comment:
I feel like this is a little hard to decipher for an end user. They wouldn't necessarily know what they missed. A fairly typical pattern is to return a list of errors from a validate method, maybe we can leverage that pattern to give more descript errors to the user.